### PR TITLE
Self-Attention GAN (SAGAN) example

### PR DESCRIPTION
**Pull request by TE-KazukiYoshiyama**
Self-Attention Generative Adversarial Networks (SAGAN) is a good GAN example since it contains many recently proposed techniques for training GANs, the following techniques are included in this SAGANs example.

- Spectral normalization: normalization of weights with its singular value to constrain the discriminator Lipchitz norm
- Two Time-Scale Update Rule (TTUR): unbalanced learning rate for the generator and discriminator
- Self-Attention: Attention of the whole region of an image
- cGANs with Projection Discriminator: projection of labels in the discriminator at the end, then use it in the adversarial loss
- Class Conditional Batch Normalaization (CCBN): Incorporation of label information into the gamma and beta of the batch normalization.

### CODE REVIEW

**Comment by TE-StefanUhlich** [comment]
@TE-KazukiYoshiyama Thanks for this nice example and all the effort you put in.
